File: backend/internal/models/monthly_allowance.go
package models

import "time"

/*
 * 〇 月次手当
 *
 * ユーザーごと・対象年月ごとの手当明細を管理する。
 * 同じユーザー・年月に複数件登録できる。
 */
type MonthlyAllowance struct {
	ID              uint       `gorm:"primaryKey" json:"id"`
	UserID          uint       `gorm:"not null;index" json:"userId"`
	TargetYear      int        `gorm:"not null;index" json:"targetYear"`
	TargetMonth     int        `gorm:"not null;index" json:"targetMonth"`
	AllowanceTypeID uint       `gorm:"not null;index" json:"allowanceTypeId"`
	Amount          int        `gorm:"not null;default:0" json:"amount"`
	Memo            string     `gorm:"type:text" json:"memo"`
	IsDeleted       bool       `gorm:"not null;default:false;index" json:"isDeleted"`
	CreatedAt       time.Time  `json:"createdAt"`
	UpdatedAt       time.Time  `json:"updatedAt"`
	DeletedAt       *time.Time `json:"deletedAt"`
}

File: frontend/src/types/admin/monthlyAllowance.ts
/*
 * 管理者 月次手当
 *
 * ユーザーごと・対象年月ごとの手当明細を管理する。
 */

export type SearchMonthlyAllowancesRequest = {
  targetUserId: number | null;
  targetYear: number | null;
  targetMonth: number | null;
  allowanceTypeId: number | null;
  includeDeleted: boolean;
  offset: number;
  limit: number;
};

export type MonthlyAllowanceDetailRequest = {
  monthlyAllowanceId: number;
};

export type CreateMonthlyAllowanceRequest = {
  targetUserId: number;
  targetYear: number;
  targetMonth: number;
  allowanceTypeId: number;
  amount: number;
  memo: string;
};

export type UpdateMonthlyAllowanceRequest = {
  monthlyAllowanceId: number;
  targetUserId: number;
  targetYear: number;
  targetMonth: number;
  allowanceTypeId: number;
  amount: number;
  memo: string;
};

export type DeleteMonthlyAllowanceRequest = {
  monthlyAllowanceId: number;
};

export type MonthlyAllowanceResponse = {
  id: number;
  userId: number;
  targetYear: number;
  targetMonth: number;
  allowanceTypeId: number;
  allowanceTypeName: string;
  amount: number;
  memo: string;
  isDeleted: boolean;
  createdAt: string;
  updatedAt: string;
  deletedAt: string | null;
};

export type SearchMonthlyAllowancesResponse = {
  monthlyAllowances: MonthlyAllowanceResponse[];
  total: number;
  offset: number;
  limit: number;
  hasMore: boolean;
};

export type MonthlyAllowanceDetailResponse = {
  monthlyAllowance: MonthlyAllowanceResponse;
};

export type CreateMonthlyAllowanceResponse = {
  monthlyAllowance: MonthlyAllowanceResponse;
};

export type UpdateMonthlyAllowanceResponse = {
  monthlyAllowance: MonthlyAllowanceResponse;
};

export type DeleteMonthlyAllowanceResponse = {
  monthlyAllowanceId: number;
};
